samplesheets/views_api.py

```diff
@@ -310,6 +317,203 @@ def post(self, request, *args, **kwargs):
         return Response(ret_data, status=status.HTTP_200_OK)
 
 
+class IrodsAccessTicketListAPIView(
+    IrodsAccessTicketModifyMixin, SODARAPIBaseProjectMixin, ListAPIView
+):
+    """
+    List iRODS access tickets for a project.
+
+    **URL:** ``/samplesheets/api/irods/ticket/list/{Project.sodar_uuid}``
+
+    **Methods:** ``GET``
+
+    **Query parameters:** ``active`` (boolean, default ``0``)
+
+    **Returns:**
+
+    - ``tickets``: List of iRODS access tickets
+    """
+
+    permission_required = 'samplesheets.edit_sheet'
+    serializer_class = IrodsAccessTicketSerializer
+
+    def get_queryset(self):
+        project = self.get_project()
+        tickets = IrodsAccessTicket.objects.filter(
+            study__investigation__project=project
+        )
+        active = self.request.query_params.get('active', '0')
+        active = bool(int(active))
+        if active:
+            tickets = [t for t in tickets if t.is_active()]
+        if not tickets:
+            raise NotFound(IRODS_TICKETS_NOT_FOUND_MSG)
+        return tickets
+
+
+class IrodsAccessTicketRetrieveAPIView(
+    IrodsAccessTicketModifyMixin, SODARAPIBaseProjectMixin, RetrieveAPIView
+):
+    """
+    Retrieve an iRODS access ticket for a project.
+
+    **URL:** ``/samplesheets/api/irods/ticket/retrieve/{IrodsAccessTicket.sodar_uuid}``
+
+    **Methods:** ``GET``
+
+    **Returns:**
+
+    - ``ticket``: iRODS access ticket
+    """
+
+    lookup_field = 'sodar_uuid'
+    lookup_url_kwarg = 'irodsaccessticket'
+    permission_required = 'samplesheets.edit_sheet'
+    serializer_class = IrodsAccessTicketSerializer
+
+    def get_object(self):
+        """Override get_object() to get IrodsAccessTicket"""
+        return IrodsAccessTicket.objects.get(
+            sodar_uuid=self.kwargs.get('irodsaccessticket')
+        )
+
+
+class IrodsAccessTicketCreateAPIView(
+    IrodsAccessTicketModifyMixin, SODARAPIBaseProjectMixin, CreateAPIView
+):
+    """
+    Create an iRODS access ticket for a project.
+
+    **URL:** ``/samplesheets/api/irods/ticket/create/{Project.sodar_uuid}``
+
+    **Methods:** ``POST``
+
+    **Parameters:**
+
+    - ``path``: iRODS path
+    - ``label``: Label (string)
+    - ``date_expires``: Expiration date (YYYY-MM-DDThh:mm:ssZ, optional)
+
+    **Returns:**
+
+    - ``ticket``: iRODS access ticket
+    """
+
+    permission_required = 'samplesheets.edit_sheet'
+    serializer_class = IrodsAccessTicketSerializer
+
+    def get_serializer_context(self):
+        context = super().get_serializer_context()
+        context['project'] = self.get_project()
+        context['user'] = self.request.user
+        return context
+
+    def perform_create(self, serializer):
+        """Override perform_create() to create IrodsAccessTicket"""
+        irods_backend = get_backend_api('omics_irods')
+        if serializer.is_valid():
```

Comment:
Validation should have happened here already, I believe this `if` clause is redundant.